helix-term/src/ui/mod.rs

```diff
@@ -255,7 +255,7 @@ pub mod completers {
             })
             .collect();
 
-        matches.sort_unstable_by_key(|(_file, score)| Reverse(*score));
+        matches.sort_unstable_by_key(|(name, score)| (Reverse(*score), name.to_lowercase()));
```

Comment:
Seems reasonable, but let's use `name` here. `to_lowercase` causes a lot of allocations

Reply:
I'm very happy if someone does that but as `name` is a `std::borrow::Cow<'_, str>` and it seems like we can't use it directly as a sort key otherwise we get:

```rust
   Compiling helix-term v0.6.0 (/home/michael/root/projects/helix/helix-term)
error: lifetime may not live long enough
   --> helix-term/src/ui/mod.rs:258:54
    |
258 |         matches.sort_unstable_by_key(|(name, score)| (Reverse(*score), name));
    |                                       -------------- ^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
    |                                       |            |
    |                                       |            return type of closure is (std::cmp::Reverse<i64>, &'2 std::borrow::Cow<'_, str>)
    |                                       has type `&'1 (std::borrow::Cow<'_, str>, i64)`

error: could not compile `helix-term` due to previous error
```

and I don't understand the error messages or the best way to work around them.

Reply:
No, cloning a `&str` allocates and gives you an owned type `String`. Taking a look, I think we have to use `sort_unstable_by` instead of `by_key`

Reply:
https://stackoverflow.com/questions/47121985/why-cant-i-use-a-key-function-that-returns-a-reference-when-sorting-a-vector-wi